src/Analysis/Core/Impl/Interpreter/PythonLibraryPath.cs

```diff
@@ -170,5 +203,90 @@ public override string ToString()
                 fs.DeleteDirectory(tempWorkingDir, true);
             }
         }
+
+        public static (IReadOnlyList<PythonLibraryPath> interpreterPaths, IReadOnlyList<PythonLibraryPath> userPaths) ClassifyPaths(
+            string root,
+            IFileSystem fs,
+            IEnumerable<PythonLibraryPath> fromInterpreter,
+            IEnumerable<string> fromUser
+        ) {
+            // PRECONDITIONS:
+            // - root has already been normalized and had its end separator trimmed.
+            // - All paths in fromInterpreter were normalised and end separator trimmed.
```

Comment:
Let's add `Debug.Assert` for this.

Reply:
Am I wrong in thinking that would have a runtime cost? I can do something like `Debug.Assert(Path.PathEquals(p, Path.NormalizePath(p)))`, but impression was that everything would be evaluated even in release mode (just not trigger anything). That being said, these operations should be rare and aren't all that expensive.

Reply:
You can wrap the whole thing in `#IF DEBUG` region. It won't be executed in Release.